erpnext/projects/doctype/project_template/project_template.js

```diff
@@ -6,3 +6,12 @@ frappe.ui.form.on('Project Template', {
 
 	// }
 });
+frappe.ui.form.on('Project Template Task', {
+	parent_task:function(frm,cdt,cdn){
+		var row = locals[cdt][cdn];
+		if(row.parent_task != null && (row.parent_task <= 0 || row.parent_task > frm.doc.tasks.length || row.parent_task === row.idx)){
```

Comment:
- I suppose the `select` field usage (mentioned below) will reduce these checks.